compiler/qsc_partial_eval/src/tests/branching.rs

```diff
@@ -1311,3 +1311,179 @@ fn if_else_expression_with_result_literal_fails() {
         ]],
     );
 }
+
+#[test]
+fn if_expression_with_classical_operand_from_hybrid_results_array_comparing_to_literal_zero() {
```

Comment:
I might be good to add one more test case that shows that arrays of mixed dynamic and static integers works as expected (I tried and it does, as that code path already handled the mixed content case). Just for completeness.

Reply:
Good point, added the missing tests for Booleans and integers.